examples/triangle/main.c

```diff
@@ -32,22 +32,40 @@
 #define RENDER_PASS_ATTACHMENTS_LENGTH (1)
 #define BIND_GROUP_LAYOUTS_LENGTH (1)
 
+typedef struct {
+    WGPUAdapterId received_adapter;
+    bool has_received_adapter;
+} RequestAdapterStatus;
+
 void request_adapter_callback(WGPUAdapterId const *received, void *userdata) {
-    WGPUAdapterId *id = (WGPUAdapterId*) userdata;
-    *id = *received;
+    RequestAdapterStatus *id = (RequestAdapterStatus*) userdata;
+    id->received_adapter = *received;
+    id->has_received_adapter = true;
 }
 
 int main() {
-    WGPUAdapterId adapter = { 0 };
+    WGPUEventLoopId event_loop = wgpu_create_event_loop();
+
+    RequestAdapterStatus request_adapter_status = {
+        .received_adapter = (WGPUAdapterId) { 0 },
+        .has_received_adapter = false
+    };
     wgpu_request_adapter_async(
         &(WGPURequestAdapterOptions){
             .power_preference = WGPUPowerPreference_LowPower,
             .backends = 2 | 4 | 8,
         },
+        event_loop,
         request_adapter_callback,
-        (void *) &adapter
+        (void *) &request_adapter_status
     );
 
+    while (!request_adapter_status.received_adapter) {
```

Comment:
We won't be able to spin on the main thread in JS like we're doing here, so we could chain callbacks instead if that's better for these native examples. We can't use wgpu-native on the web anyway so I don't have much of a preference.

In wgpu-rs, we can chain callbacks or use `Future`s (mapping `Promise`->`Future` for web, and decide on some strategy to map callbacks -> `Future`s for native)

Reply:
That's a little bit tricky. We can't use wgpu-native on the web, but it would be very desired for the native examples to be written in a way that fits both web and non-web scenarios.

Reply:
Moved this into a callback instead